### PR TITLE
feat: add config_get and tests

### PR DESCRIPTION
**Pull request by mhmd-azeez**

### CODE REVIEW

**Comment by mhmd-azeez** [comment]
It seems like none of hte bindgens used in this repo can handle nullable output:

- TS:

```tsx
/**
 * Get the configuration for the tool
 *
 * @param {string} input - The config key
 * @returns {string | null} The config value for the given key, or empty if not found
 */
export function config_get(input: string): string | null {
  const mem = Memory.fromString(input as string);

  const ptr = hostFunctions.config_get(mem.offset);

  return Memory.find(ptr).readJsonObject();
}

```

```go
Panic at call: Uncaught TypeError: cannot read property 'readJsonObject' of undefined
    at config_get (script.js:128)
    at callImpl (script.js:137)
    at call (script.js:168)
```

- Rust:

```rust
/// config_get Get the configuration for the tool
/// It takes input of String (The config key)
/// And it returns an output Option<String> (The config value for the given key, or empty if not found)
#[allow(unused)]
pub(crate) fn config_get(input: String) -> std::result::Result<Option<String>, extism_pdk::Error> {
    let res = unsafe { raw_imports::config_get(input)? };

    let extism_pdk::Json(res) = res;

    Ok(res)
}

```

```go
Panic at call: EOF while parsing a value at line 1 column 0.
```

- C++:
```cpp
EXTISM_IMPORT_USER("config_get")
extern extism::imports::RawHandle config_get(extism::imports::RawHandle);

} // namespace imports

std::expected<std::string, Error> config_get(std::string_view input) {
  auto &encoded = input;
  auto in_handle = extism::UniqueHandle<char>::from(encoded);
  if (!in_handle) {
    return std::unexpected(Error::extism);
  }
  auto out_raw = imports::config_get(*in_handle);
  if (!out_raw) {
    return std::unexpected(Error::host_null); // <-- fails here
  }
  extism::UniqueHandle<char> out_handle(out_raw);
  auto out_string = out_handle.string();
  if (!out_string.size()) {
    return std::unexpected(Error::not_json);
  }
  return jsoncons::decode_json<std::string>(std::move(out_string));
}
```

File: justfile [hunk 1]
```diff
@@ -18,3 +18,14 @@ push:
...
 
+test:
+  #!/usr/bin/env sh
+  xtp plugin build --path test/host
+  xtp plugin build --path test/testsuite
+
+  for dir in servlets/*/; do
+    cd "$dir"
+    echo "Testing $dir"
+    xtp plugin test --allow-host api.fxratesapi.com --allow-host getxtp.com --log-level warn
```
Comment (bhelx) [inline justfile:29]: we might want to consider just doing`--allow-host "*"` for these.

Comment (mhmd-azeez) [reply justfile:29]: xtp test doesnt have configuration for allowedPaths, so we cant test them yet

**Comment by mhmd-azeez** [comment]
closed in favor of https://github.com/dylibso/mcp.run-servlets/pull/9

File: test/host/main.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	"fmt"
+
+	pdk "github.com/extism/go-pdk"
+)
+
+//go:wasmimport xtp:test/harness mock_input
+func mock_input() uint64
+
+//go:export config_get
+func config_get(key uint64) uint64 {
```
Comment (nilslice) [inline test/host/main.go:14]: did @bhelx get rid of this from the schema?

File: justfile (hunk 1 above)
Comment (nilslice) [reply justfile:29]: we can use the Extism CLI too in a pinch

File: test/host/main.go (hunk 2 above)
Comment (mhmd-azeez) [reply test/host/main.go:14]: yes, the prod schema no longer has the `config_get` import

**Pull request closed without merging**